_posts/2018-08-30-pair-programming.markdown

```diff
@@ -0,0 +1,184 @@
+avoid a lot of "I wasn't sure about X" moments new hires might have encountered. At the same time, as mentioned above,
+pair-programming is not a silver bullet. It is important to be able to use pair-programming as a tool in your toolbox
+to solve a particular issue in your engineering team.
+
+## What You'll Need to Pair Program
+
+The minimum you need is a computer and the ability to communicate, but let's look at what an optimal pair
+programming setup looks like.
+
+### Offline
+
+If you can have physical access to the other engineer, then you should aim to have:
+
+- One computer set up as a workstation.
+- Two sets of keyboards, mice and monitors attached.
+- A spare computer for researching on the side.
+
+This gives you the ability to iterate and switch responsibilities at the speed of thought. If you have the resources
+and space, then setting up a place where any 2 engineers can drop a laptop down and be productive instantly will
+reduce a lot of the setup friction.
+
+### Remote
+
+The minimum you would need in this case is a video chat system like Skype, Google Hangouts or Slack Calls. Given
+that all of these support screen sharing too, then you can easily replicate two people sitting next to each other
+with one computer.
+
+It's worth noting that remote pairing requires stable internet, and with high bandwidth. We've both had situations
+in the past where we've had to abort pairing sessions due to internet un-reliability.
+
+We used to use ScreenHero a lot, to support multiple mice and keyboards, but [ScreenHero was bought by slack][sh]
+and that feature still isn't available for everyone. So, a lot of our pairing is done via screen sharing.
+
+There's a newcomer to the scene though, [VS Code's LiveShare][ls] gives you the ability to share an IDE, do voice
+chat, server ports and terminal sessions with a very low amount of setup. We wrote the initial draft of this post
+in-person, in aN Artsy meeting room writing and talking in real-time. If you've not seen Live Share, we posted a
+video of a [workshop we ran][ls-yt] at Artsy on YouTube.
+
+If you're looking to find more resources on pair programming, Joe Moore's: [Remote Pair Programming][rpp] is a great
+place to start. He's been working at Pivotal Labs for over a decade.
+
+### Attribution
+
+One way to encourage pair programming is by providing insight to others about when you're working as a pair. A great
+technique for this is to use [`git-duet`][git-duet]. Git Duet is a tool which extends git to make it easy to share
+attribution among contributors. It does this by having a central list of people' emails, you set up a pairing
+session, then use `git duet-commit` to replace `git commit` and the attribution shared.
+
+You can get started with:
+
+```sh
+$ brew tap git-duet/tap
+$ brew install git-duet
+```
+
+You'll then need to set up a `~/.git-authors` file which is a map of people's names to emails:
+
+```yml
+pairs:
```

Comment:
I like that you've included an actual list, but should we include engineers who no longer work at Artsy? I guess you might still find yourself pairing with them someday 😄

Reply:
I removed Craig from this example 😢